src/main/java/htsjdk/samtools/DuplicateScoringStrategy.java

```diff
@@ -87,9 +91,23 @@ public static short computeDuplicateScore(final SAMRecord record, final ScoringS
                     }
                     break;
                 case RANDOM:
-                    score += (short) (hasher.hashUnencodedChars(record.getReadName()) >> 16);
+                    // start with a random number between Short.MIN_VALUE/4 and Short.MAX_VALUE/4
+                    score += (short) (hasher.hashUnencodedChars(record.getReadName()) & 0b11_1111_1111_1111);
+                    // subtract Short.MIN_VALUE/4 from it to end up with a number between
+                    // 0 and Short.MAX_VALUE/2. This number can be then discounted in case the read is
+                    // not passing filters. We need to stay far from overflow so that when we add the two
+                    // scores from a read-pair we do not overflow since that could cause us to chose the
+                    // Failing read-pair instead of a passing one.
+                    score -= Short.MIN_VALUE/4;
             }
 
+            // make sure that filter-failing records are heavily discounted. Dividing by 2 to be far from
+            // overflow when adding score from both fragments of a read-pair
+
+            // a long read with high quality bases may overflow and
+            score = (short) Math.max(Short.MIN_VALUE, score);
+            score += record.getReadFailsVendorQualityCheckFlag() ? (short) (Short.MIN_VALUE / 2) : 0;
```

Comment:
Also, from reading the comment I thought you were just going to do `score = score / 2`, but you're not.  Either the comment needs to explain a little better, or the code needs to just divide the score.